open-apparel-registry/open-apparel-registry
# Add extended fields to sidebar

## Overview

Add extended fields to the sidebar search form to enable filtering facilities on extended fields.
 - **Parent Company**: Use react-select `creatable` to specify one ore more contributors. The data source for the select list is the same as the "Filter by Contributor" field. If one of these is selected, set the query string argument to the
numeric ID; If the item is not in the list, set the query string argument to the text name. Update list with label to make sure parent company name instead of id shows up when accessing through url
 - **Facility Type** and **Processing Type**: Make request to fetch the facility/processing type choice values, and parse the response to populate the facility type and processing type multi-select list. The two multi-select lists are connected - If one or more facility_types is selected, only allows selecting processing types that belong to one of those types, and vice versa.
 - **Product Type**: Use the standard product types and custom ones from the claimed
facilities to populate the options. Use react-select `creatable` to
specify one ore more potentially multi-word search values
 - **Number of Workers**: Get the value from the workers_range values to populate the select
options.
 - **Native Language Name**

Connects #1589 

## Demo

https://user-images.githubusercontent.com/60887686/158299941-8fa622b1-6045-4f33-ac21-f93f5172b974.mov

## Notes

 - Partial text search for `parent_company` will be implemented in a follow-up card
 - The UI doesn't exactly match the [design](https://www.figma.com/file/x4ae8NNWyGfSo7spUvH5u1/Open-Apparel-Registry?node-id=58%3A798) due to the filter by area is moving
 - There are two `Embellishment` and `Embroidery` under two different facility types according to [taxonomy](https://docs.google.com/spreadsheets/d/1HlGoYgj0rbtxnhgm0LAWk5lgFCSw3A2V/edit#gid=299419561). This may cause confusion for users but not sure about the best way to address
![image](https://user-images.githubusercontent.com/60887686/158300810-3e18e27d-23ec-4771-83be-163998ce0115.png)

## Testing Instructions

* Checkout this branch and run `server`	
* Contribute [ExtendedFieldsTestList.csv](https://github.com/open-apparel-registry/open-apparel-registry/files/8249947/ExtendedFieldsTestList.csv) or other extended field lists and fully process it using `./tools/batch_process {list-id}`
* Expand the sidebar to view more filters, put in different kinds of value to make sure all fields are working as intended individually and together
	 - The options in facility type and processing type should make sense - they should always be under the same facility/processing type group
	 - The two creatable select list (parent company and product type) should search on both options and custom text
 * Paste the url to a new browser tab
	 - When any extended fields are included in the url, the sidebar should expand when loading

## Checklist
